docs/aws/audit/iamcompliance/rules/user_account_access_keys_rotation.mdx

```diff
@@ -23,6 +23,271 @@ HIPAA, GDPR, CISAWS, CBP, NIST, HITRUST
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. Install and configure AWS CLI: Before you can start using AWS CLI, you need to install it on your local machine. You can download it from the official AWS website. After installation, you need to configure it with your AWS account credentials. You can do this by running the command `aws configure` and then entering your AWS Access Key ID, Secret Access Key, Default region name, and Default output format when prompted.
+
+2. List all IAM users: To check the access keys for all IAM users, you first need to list all the IAM users. You can do this by running the command `aws iam list-users`. This command will return a list of all IAM users in your AWS account.
+
+3. List access keys for each user: For each user, you need to list their access keys to check when they were last rotated. You can do this by running the command `aws iam list-access-keys --user-name <username>`. Replace `<username>` with the name of the user. This command will return a list of all access keys for the specified user, along with their creation dates.
+
+4. Check the age of each access key: To check the age of each access key, you need to compare the creation date of the key with the current date. You can do this manually, or you can write a script to automate the process. If the age of the key is more than 90 days, it is recommended to rotate the key.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install the necessary Python libraries: Before you start, you need to install the AWS SDK for Python (Boto3) in your environment. You can do this using pip:
+
+   ```
+   pip install boto3
+   ```
+
+2. Import the necessary libraries and establish a session: In your Python script, you'll need to import the Boto3 library and establish a session with your AWS account. You'll need your AWS access key ID and secret access key.
+
+   ```python
+   import boto3
+   from datetime import datetime, timedelta
+
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
+       region_name='us-west-2'
+   )
+   ```
+
+3. Retrieve IAM users and their access keys: Use the IAM client in Boto3 to retrieve a list of all IAM users and their access keys. 
+
+   ```python
+   iam = session.client('iam')
+
+   users = iam.list_users()['Users']
+   for user in users:
+       access_keys = iam.list_access_keys(UserName=user['UserName'])['AccessKeyMetadata']
+   ```
+
+4. Check the age of each access key: For each access key, check the 'CreateDate' attribute to determine how old the key is. If the key is older than 90 days, print a warning message.
+
+   ```python
+   for key in access_keys:
+       create_date = key['CreateDate']
+       age = datetime.now(create_date.tzinfo) - create_date
+       if age > timedelta(days=90):
+           print(f"Warning: Access key {key['AccessKeyId']} for user {user['UserName']} is {age.days} days old.")
```

Comment:
<h2>Secret Detected: Secret Keyword</h2> <a href="https://console.cloudanix.com/code?t=secrets"> Click here for more details</a>